design.md

```diff
@@ -1,13 +1,42 @@
 # Design
+My language, in a nutshell, is a reimagining of picobot as a car on a roadtrip, so the language instructions are meant to be reminiscent of instructions from a GPS.
```

Comment:
I think that this is a really creating reimagining of picobot! I also think you made the right call writing this as an external DSL rather than an internal one; what you wanted to do here would not have played nice within the bounds of Scala!